common/_run_frescox_simulation.py

```diff
@@ -20,39 +19,55 @@
 MPI_N_PROCESSES = "n_processes"
 
 
-def run_frescox_simulation(frescox, config, mpi_setup, filename, overwrite):
+def _run_frescox_simulation(
+    frescox: dict,
+    config: Configuration,
+    mpi_setup: Optional[dict],
+    filename: Path,
+    overwrite: bool,
+    cwd: Optional[Path] = None,
+):
```

Comment:
Why is `mpi_setup` marked as optional?  If it is, shouldn't a default value be provided?  Also, shouldn't it be moved to the end of the argument list with `cwd`?

Since this is an important, internal function, my preference is that this interface avoid optional arguments so that the code in the packages' `run_simulation` functions be forced to explicitly make decisions on all arguments.

While the use of an optional `cwd` is a bit more acceptable, in the same spirit you could force the `run_simulation` to supply an actual value.  This is OK because `cwd` can be optional in the public `run_simulation` functions if so desired.

Also, filename can be a Path or a str.